src/SablierMerkleVCA.sol

```diff
@@ -74,36 +75,8 @@ contract SablierMerkleVCA is
             params.token
         )
     {
-        // Check: vesting start time is not zero.
-        if (params.vestingStartTime == 0) {
-            revert Errors.SablierMerkleVCA_StartTimeZero();
-        }
-
-        // Check: vesting end time is greater than the vesting start time.
-        if (params.vestingEndTime <= params.vestingStartTime) {
-            revert Errors.SablierMerkleVCA_VestingEndTimeNotGreaterThanVestingStartTime({
-                vestingStartTime: params.vestingStartTime,
-                vestingEndTime: params.vestingEndTime
-            });
-        }
-
-        // Check: campaign expiration is not zero.
-        if (params.expiration == 0) {
-            revert Errors.SablierMerkleVCA_ExpirationTimeZero();
-        }
-
-        // Check: campaign expiration is at least 1 week later than the vesting end time.
-        if (params.expiration < params.vestingEndTime + 1 weeks) {
-            revert Errors.SablierMerkleVCA_ExpirationTooEarly({
-                vestingEndTime: params.vestingEndTime,
-                expiration: params.expiration
-            });
-        }
-
-        // Check: unlock percentage is not greater than 100%.
-        if (params.unlockPercentage.unwrap() > uUNIT) {
-            revert Errors.SablierMerkleVCA_UnlockPercentageTooHigh(params.unlockPercentage);
-        }
+        // Peform the checks.
+        ISablierFactoryMerkleVCA(msg.sender).computeMerkleVCA(campaignCreator, params);
```

Comment:
I think we should move these checks to the Factory contract and have no revert conditions inside this constructor. Initially we had these checks placed in constructor so that the contract cannot be deployed with incorrect parameters directly. But because now its making a callback to `msg.sender`, it can be deployed with incorrect parameters if `msg.sender` is not our factory. Therefore, I do not see any value in placing it here.

Moving it to constructors solves two issues:

1. The campaign contract can be deployed without any revert. That means, if user pre funds the address, he will be able to recover them.
2. Cleaner code.